backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -361,24 +475,48 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
       case other => s"The Cache hit copying actor timed out waiting for an unknown I/O operation: $other"
     }
 
-    // Loggable because this is an attempt-specific:
+    // Loggable because this is attempt-specific:
     failAndStop(LoggableCacheCopyError(new TimeoutException(exceptionMessage)))
     ()
   }
 
   /**
-    * If a subclass of this `StandardCacheHitCopyingActor` supports blacklisting then it should implement this
+    * If a subclass of this `StandardCacheHitCopyingActor` supports blacklisting by path then it should implement this
     * to return the prefix of the path from the failed copy command to use for blacklisting.
     */
   protected def extractBlacklistPrefix(path: String): Option[String] = None
 
   private def sourcePathFromCopyOutputsCommand(command: CopyOutputsCommand): String = command.jobDetritusFiles.values.head
 
+  // The "published*BlacklistRead" vars are to deal with the evaluation of the partial function that uses these
+  // `isSourceBlacklisted` methods in its guards. There are both "isDefinedAt and "apply" invocations associated with
+  // the partial function which means this code gets called twice. If the code was pure that would be fine, but
+  // publishing metrics is obviously side-effecting so these vars prevent the same metric from being published
+  // multiple times.
+
+  //noinspection ActorMutableStateInspection
+  private var publishedBucketBlacklistRead = false
+
   private def isSourceBlacklisted(command: CopyOutputsCommand): Boolean = {
     val path = sourcePathFromCopyOutputsCommand(command)
     (for {
+      cache <- standardParams.blacklistCache
       prefix <- extractBlacklistPrefix(path)
+      value = cache.getBlacklistStatus(prefix)
+      _ = if (!publishedBucketBlacklistRead) publishBlacklistMetric(cache, verb = "read", bucketOrHit = "bucket", prefix, value)
```

Comment:
This `publishedBucketBlacklistRead` is one-per-`StandardCacheHitCopyingActor`? So this would only ever log once per actor instance? 

Oh, are these one-use actors?

Reply:
yup one use actors